lighthouse-core/runner.js

```diff
@@ -328,17 +328,19 @@ class Runner {
         ...sharedAuditContext,
       };
 
-      // Only pass the declared `requiredArtifacts` to the audit
+      // Only pass the declared `requiredArtifacts`/`optionalArtifacts` to the audit
       // The type is masquerading as `LH.Artifacts` but will only contain a subset of the keys
       // to prevent consumers from unnecessary type assertions.
-      const requiredArtifacts = audit.meta.requiredArtifacts
-        .reduce((requiredArtifacts, artifactName) => {
+      const requestedArtifacts = audit.meta.requiredArtifacts
+        .concat(audit.meta.optionalArtifacts || []);
+      const narrowedArtifacts = requestedArtifacts
+        .reduce((narrowedArtifacts, artifactName) => {
           const requiredArtifact = artifacts[artifactName];
```

Comment:
while we're incurring all the rename churn anyway... :)
```suggestion
          const requestedArtifact = artifacts[artifactName];
```